### DISCUSSION TITLE
Anomaly detection with an array of labels?

### DISCUSSION THREAD

**Discussion by petteriTeikari** [aeon-toolkit/aeon#2328] [Q&A]
Hi,

I am new to `aeon` and stumbled on this as [`UniTS`](https://github.com/mims-harvard/UniTS/blob/main/Tutorial.md) suggested converting my own data to `.ts` format. I could not quickly see how can you supply an array of labels to anomaly detection?

e.g. in the tutorial:
https://github.com/aeon-toolkit/aeon-tutorials/blob/main/ECML-2024/Notebooks/part6_anomaly_detection.ipynb

or in the desired tutorial issue:
https://github.com/aeon-toolkit/aeon/issues/1960

I have for example `(16, 1, 1981)` input data array with the 16 subjects being totally independent, and having their outliers happening at different times and cannot give a `y` size of `(1981,)` but would like to give `(16,1981)` type of  shape.

Or do I simply just concatenate the array to `(1,1,1981*16)` shape with `y` with shape `(1981*16)`

Thanks

**Comment by SebastianSchmidl** [comment]
Is this question about (1) storing anomaly detection data, including the anomaly scores/labels, to disk using the `.ts`-file format, or (2) using estimators from `aeon.anomaly_detection` to detect anomalies in multiple time series?

1. You cannot store the anomaly labels/scores (of shape `(n_timesteps,)`) in a `.ts`-file, we currently use the [TimeEval file format](https://timeeval.readthedocs.io/en/stable/concepts/datasets.html#canonical-file-format).
2. The estimators in `aeon.anomaly_detection` are all series estimators and do not support collections. You need to call the estimator's `predict` for each time series individually.
